snuba/clickhouse/formatter.py

```diff
@@ -54,30 +54,41 @@ def __alias(self, formatted_exp: str, alias: Optional[str]) -> str:
 
     def visit_literal(self, exp: Literal) -> str:
         if exp.value is None:
-            return "NULL"
+            return self.__alias("NULL", exp.alias)
         elif exp.value is True:
-            return "true"
+            return self.__alias("true", exp.alias)
         elif exp.value is False:
-            return "false"
+            return self.__alias("false", exp.alias)
         elif isinstance(exp.value, str):
-            return escape_string(exp.value)
+            return self.__alias(escape_string(exp.value), exp.alias)
         elif isinstance(exp.value, (int, float)):
-            return str(exp.value)
+            return self.__alias(str(exp.value), exp.alias)
         elif isinstance(exp.value, datetime):
             value = exp.value.replace(tzinfo=None, microsecond=0)
-            return "toDateTime('{}', 'Universal')".format(value.isoformat())
+            return self.__alias(
+                "toDateTime('{}', 'Universal')".format(value.isoformat()), exp.alias
+            )
         elif isinstance(exp.value, date):
-            return "toDate('{}', 'Universal')".format(exp.value.isoformat())
+            return self.__alias(
+                "toDate('{}', 'Universal')".format(exp.value.isoformat()), exp.alias
+            )
         else:
             raise ValueError(f"Unexpected literal type {type(exp.value)}")
 
     def visit_column(self, exp: Column) -> str:
         ret = []
+        ret_unescaped = []
         if exp.table_name:
             ret.append(escape_identifier(exp.table_name) or "")
+            ret_unescaped.append(exp.table_name or "")
             ret.append(".")
+            ret_unescaped.append(".")
         ret.append(escape_identifier(exp.column_name) or "")
-        return self.__alias("".join(ret), exp.alias)
+        ret_unescaped.append(exp.column_name)
+        if exp.alias != "".join(ret_unescaped):
+            return self.__alias("".join(ret), exp.alias)
```

Comment:
Just a minor optimization for queries readability. There is no reason to apply aliases to columns when the alias name is the same as the column name `(group_id AS group_id)`

Reply:
Curious why add this optimization? I assume it never happens since since the query language doesn't allow for the user to pass aliases on columns anyway.

Reply:
The parser applies column aliases by default so that the query processors have an easier life. So it actually happens 100% of the times.

An example
```
{
    "group": "GROUP BY (`tags[sentry:release]`)",
    "prewhere": "PREWHERE in((nullIf(group_id, 0) AS group_id), tuple(1727529708))",
    "limit": "LIMIT 1 OFFSET 0",
    "from": "FROM sentry_dist",
    "where": "WHERE and(equals(deleted, 0), and(greaterOrEquals(timestamp, toDateTime('2020-06-15T15:00:09', 'Universal')), and(less(timestamp, toDateTime('2020-06-16T15:10:13', 'Universal')), and(in(project_id, tuple(190580)), and(isNotNull(`tags[sentry:release]`), in((project_id AS project_id), tuple(190580)))))))",
    "order": "ORDER BY seen ASC",
    "select": "SELECT (`sentry:release` AS `tags[sentry:release]`), (min((timestamp AS timestamp)) AS seen)"
}
```

Those aliases are applied so that query processor just have to preserve them if they transform a column into something else.

Reply:
Ok. can you add the comment about the optimization to the code?

Reply:
done